### PR TITLE
[Community]: AzureSearchVectorStoreRetriever Fixed to account for search_kwargs

### PR DESCRIPTION
**Pull request by keenborder786**
- **Description:** Fixed `AzureSearchVectorStoreRetriever` to account for search_kwargs. More explanation is in the mentioned issue.
- **Issue:** #21492

### CODE REVIEW

**Comment by keenborder786** [comment]
@baskaryan  please review

**Comment by keenborder786** [comment]
@mspronesti I have made your changes. Please check

**Comment by mspronesti** [comment]
@keenborder786 Thanks for taking my suggestion into consideration -- guess you forgot to apply these changes in the last 3 `elif` branches.

**Comment by keenborder786** [comment]
@baskaryan, @efriis, @eyurtsev, @hwchase17 please review

File: libs/community/langchain_community/vectorstores/azuresearch.py [hunk 1]
```diff
@@ -750,26 +761,32 @@ def _get_relevant_documents(
...
                 doc
                 for doc, _ in self.vectorstore.hybrid_search_with_relevance_scores(
-                    query, k=self.k, **kwargs
+                    query, k=self.k, **self.search_kwargs, **kwargs
                 )
             ]
         elif self.search_type == "semantic_hybrid":
-            docs = self.vectorstore.semantic_hybrid_search(query, k=self.k, **kwargs)
+            docs = self.vectorstore.semantic_hybrid_search(
+                query, k=self.k, **self.search_kwargs, **kwargs
```
Comment (mspronesti) [inline libs/community/langchain_community/vectorstores/azuresearch.py:788]:
```suggestion
                query, k=self.k, **params
```

**Comment by mspronesti** [comment]
@keenborder786 There's a small conflict which arose from my latest PR being merged. Resolving it should only require replacing `kwargs` with `params` in the new search type `elif` branch.

@ccurme  Could you please review this as well? :)

**Comment by keenborder786** [comment]
> @keenborder786 There's a small conflict which arose from my latest PR being merged. Resolving it should only require replacing `kwargs` with `params` in the new search type `elif` branch.
> 
> @ccurme Could you please review this as well? :)

okay, I will resolve it.

**Comment by keenborder786** [comment]
@mspronesti  fixed

**Comment by keenborder786** [comment]
@eyurtsev please see

**Review by baskaryan — DISMISSED** [review]

File: libs/community/langchain_community/vectorstores/azuresearch.py [hunk 2]
```diff
@@ -740,6 +739,18 @@ class AzureSearchVectorStoreRetriever(BaseRetriever):
     or "semantic_hybrid_score_threshold"."""
     k: int = 4
     """Number of documents to return."""
+    search_kwargs: dict = Field(
+        default_factory=lambda: {"fetch_k": 20, "lambda_mult": 0.5}
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/azuresearch.py:743]: can we just default to empty dict here and let vecstore set defaults?

Comment (keenborder786) [reply libs/community/langchain_community/vectorstores/azuresearch.py]: @baskaryan done

**Review by baskaryan — APPROVED** [review]

**Pull request merged**